src/wallet/wallet2.cpp

```diff
@@ -2720,7 +2720,10 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
           short_chain_history.clear();
           get_short_chain_history(short_chain_history);
           fast_refresh(stop_height, blocks_start_height, short_chain_history, true);
-          THROW_WALLET_EXCEPTION_IF(m_blockchain.size() != stop_height, error::wallet_internal_error, "Unexpected hashchain size");
+          fast_refresh(stop_height, blocks_start_height, short_chain_history, true);
+          if (m_blockchain.size() != stop_height || (m_blockchain.size() != 1 && stop_height != 0)) {
```

Comment:
That seem wrong. It throws in *more* cases, rather than not throw in the special case.

Reply:
fixed.